auto_nag/scripts/regressor.py

```diff
@@ -0,0 +1,96 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import os
+from auto_nag import logger
+from auto_nag.bugbug_utils import BugbugScript
+from auto_nag.utils import get_login_info, nice_round
+from bugbug.models.regressor import RegressorModel
+from bugbug import db, repository
+import hglib
+
+
+class Regressor(BugbugScript):
+    def __init__(self):
+        self.model_class = RegressorModel
+
+        self.repo_dir = get_login_info()["repo_dir"]
+
+        if not os.path.exists(self.repo_dir):
+            cmd = hglib.util.cmdbuilder(
+                "robustcheckout",
+                "https://hg.mozilla.org/mozilla-central",
+                self.repo_dir,
+                purge=True,
+                sharebase=self.repo_dir + "-shared",
+                networkattempts=7,
+                branch=b"tip",
+            )
+
+            cmd.insert(0, hglib.HGPATH)
+
+            proc = hglib.util.popen(cmd)
+            out, err = proc.communicate()
+            if proc.returncode:
+                raise hglib.error.CommandError(cmd, proc.returncode, out, err)
+
+            logger.info("mozilla-central cloned")
+
+            # Remove pushlog DB to make sure it's regenerated.
```

Comment:
don't you need to get vcs too and set up a hgrc file ?

Reply:
We can just install Mercurial and set up hgrc on the VM, or how do we do that without Docker?

Reply:
my bad I thought version-control-tools (vct) and I wrote vcs.

Reply:
How do we get that? We still need to have it in the VM, no? It could be a manual setup step

Reply:
You can clone the repo as you're doing for hgmo and then create a hgrc.
But yep I can do that locally.
And it's really time to have a Docker: I'll try to do that next week.

Reply:
Yeah I think Docker would be safer here, as otherwise if we have multiple things using hgrc we could create conflicts.
This is the Docker file we have for bugbug commit retrieval task: https://github.com/mozilla/bugbug/blob/master/infra/dockerfile.commit_retrieval#L3, there are some things in there which could be useful when you do the Docker image for autonag.